qutebrowser/browser/greasemonkey.py

```diff
@@ -146,15 +157,27 @@ class GreasemonkeyManager(QObject):
 
     def __init__(self, parent=None):
         super().__init__(parent)
+        self._run_start = []
+        self._run_end = []
+        self._run_idle = []
+        self._in_progress_dls = []
+        self._pending_dls = []
+
         self.load_scripts()
+        app = objreg.get('app')
+        app.new_window.connect(self._start_pending_downloads)
```

Comment:
Can you elaborate on this? Why do you need to call this when a new window appears?

Reply:
The `qtnetwork-download-manager` that I'm using is only initialized in MainWindow, the greasemonkey init runs earlier than that. So I only care that there is a window, I could detach that signal listener the first time it is called, probably.
Why is there two download managers instead of one abstract interface?

Reply:
Because with QtWebEngine, downloads initiated via the website uses the QtWebEngine downloads, while other custom downloads (adblocking, etc.) can't use that and need to use the QtNetwork one (also used for QtWebKit) instead.

What I'm wondering though is why there isn't just one global download manager - currently the networkmanager associated with it needs a window ID, but maybe it's possible to change that...